src/utils.jl

```diff
@@ -9,6 +9,38 @@ macro check_args(D, cond)
     end
 end
 
+isuncountable(x) = false
```

Comment:
I would be for adding only one of `isuncountable`, `iscountable`, given that one is the negation of the other.
`iscountablereal` should also not be necessary, and can rely on `iscountable` and type dispatch

Reply:
Actually, there's a problem there.

Nearly all distributions cannot currently handle not-`Float64` eltypes, but will be able to handle `<: Real` eltypes with only a modest amount of work. However, it's much harder to get them to handle (un)countable non-real `<: Number` eltypes, such as Unitful types. Nonetheless, the aspiration should be to handle all of these types, so I want to be able to move easily from the `iscountablereal(T)` to `iscountable(T)` checks which have been created (`@check_countable()`, etc.).

Similarly, `iscountable(T)` is not just `!isuncountable(T)`. Non-`Number` types are rejected by both for now, as they are well beyond the scope of the current series of PRs. I could call them `iscountablenumbner()` and `isuncountablenumber()` to make it clear that they are not inverses?

Reply:
PS The other option I thought of (as you suggest) was to have type dispatch on the macro for the time being (which is the only user of `isuncountablereal()`, and this function is also not exported incidentally), but I assume that's still not possible?

Reply:
Okay, I've got rid of this conflict - I've added three level logic - `missing` - for the cases where we haven't yet resolved whether it's countable or uncountable, then the two are inverses of each other.